source/jormungandr/jormungandr/interfaces/v1/JourneyCommon.py

```diff
@@ -0,0 +1,244 @@
+
+        # We set default modes for fallback modes.
+        # The reason why we cannot put default values in parser_get.add_argument() is that, if we do so,
+        # fallback modes will always have a value, and traveler_type will never override fallback modes.
+        if args.get('origin_mode') is None:
+            args['origin_mode'] = ['walking']
+        if args.get('destination_mode') is None:
+            args['destination_mode'] = ['walking']
+
+        #we transform the origin/destination url to add information
+        if args['origin']:
+            args['origin'] = transform_id(args['origin'])
+        if args['destination']:
+            args['destination'] = transform_id(args['destination'])
+        if not args['datetime']:
+            args['datetime'] = args['_current_datetime']
+
+        if args['data_freshness'] is None:
+            # retrocompatibilty handling
+            args['data_freshness'] = \
+                'adapted_schedule' if args['disruption_active'] is True else 'base_schedule'
+
+        # for last and first section mode retrocompatibility
+        if 'first_section_mode' in args and args['first_section_mode']:
+            args['origin_mode'] = args['first_section_mode']
+        if 'last_section_mode' in args and args['last_section_mode']:
+            args['destination_mode'] = args['last_section_mode']
+
+        # TODO : Changer le protobuff pour que ce soit propre
+        if args['destination_mode'] == 'vls':
+            args['destination_mode'] = 'bss'
+        if args['origin_mode'] == 'vls':
+            args['origin_mode'] = 'bss'
+
+        if region:
+            self.region = i_manager.get_region(region)
+            if uri:
+                objects = uri.split('/')
+                if objects and len(objects) % 2 == 0:
+                    args['origin'] = objects[-1]
+                else:
+                    abort(503, message="Unable to compute journeys "
+                                   "from this object")
+
+        def _set_specific_params(mod):
+            if args.get('_walking_transfer_penalty') is None:
+                args['_walking_transfer_penalty'] = mod.walking_transfer_penalty
+            if args.get('_night_bus_filter_base_factor') is None:
+                args['_night_bus_filter_base_factor'] = mod.night_bus_filter_base_factor
+            if args.get('_night_bus_filter_max_factor') is None:
+                args['_night_bus_filter_max_factor'] = mod.night_bus_filter_max_factor
+            if args.get('_max_additional_connections') is None:
+                args['_max_additional_connections'] = mod.max_additional_connections
+
+        if region:
+            _set_specific_params(i_manager.instances[region])
+        else:
+            _set_specific_params(default_values)
+
+        if not region:
+            #TODO how to handle lon/lat ? don't we have to override args['origin'] ?
+            self.possible_regions = compute_regions(args)
```

Comment:
I'm not a fan of changing the list of possible_regions by side effect, I think it's better to return this list, and use the returned value